### DISCUSSION TITLE
Getting LFS objects and sizes in all repos in an Org

### DISCUSSION THREAD

**Discussion by svanimisetti** [community/community#144860] [Apps, API and Webhooks]
I would like to use GitHub API to get a list of all LFS objects, pointer files, size and link to the LFS blob. The only LFS- specific endpoints I only found were to enable or disable LFS in a repo [using LFS REST endpoints](https://docs.github.com/en/enterprise-cloud@latest/rest/repos/lfs?apiVersion=2022-11-28). One possible approach is to get all files using [contents endpoint](https://docs.github.com/en/rest/repos/contents?apiVersion=2022-11-28), manually check each file if it's an LFS pointer file, and then process the information from there. I think this is inefficient and cumbersome process, and am certain that there are better ways to do this. Any guidance or suggestions will be really appreciated!

**Comment by IRNCyber** [comment]
Unfortunately, GitHub's public API doesn't directly provide a comprehensive list of LFS objects, pointers, and their associated blobs.
The approach you mentioned, while technically feasible, is indeed inefficient and cumbersome. It would require you to fetch every file in the repository, check if it's an LFS pointer, and then parse the pointer to get the LFS object information.
Here are some alternative approaches you might consider:
 * GitHub Enterprise API:
   * If you have access to a GitHub Enterprise instance, it might offer more granular API endpoints, including LFS-specific ones. Check your organization's specific API documentation.
 * Git LFS CLI:
   * You could use the Git LFS command-line interface to:
     * Clone the repository locally.
     * Use git lfs ls-files to list all LFS objects.
     * Parse the output to extract the necessary information.
     * This approach might be more suitable for one-time or occasional queries.
 * Third-Party Tools and Libraries:
   * Explore third-party tools and libraries that interact with GitHub's API and provide more robust LFS object information. Some tools might offer specific functionalities for LFS data extraction.
Important Considerations:
 * Rate Limits: Be mindful of GitHub's API rate limits, especially when querying large repositories or making frequent requests.
 * Authentication: You'll need to authenticate your requests using a personal access token or OAuth app.
 * Error Handling: Implement robust error handling to handle potential API errors and rate limit issues.
 * Performance Optimization: Consider batching requests, caching results, and using pagination to optimize performance.
While the direct approach using GitHub's public API might be limited, combining these alternative methods and careful API usage can help you efficiently extract the desired LFS object information.
Would you like to provide more context about your specific use case? Knowing more about your goals can help me suggest a more tailored approach. For instance, if you're looking to analyze LFS usage patterns or optimize repository size, there might be additional strategies to consider.
